src/coreclr/src/jit/morph.cpp

```diff
@@ -10604,15 +10596,30 @@ GenTree* Compiler::fgMorphCopyBlock(GenTree* tree)
             // Are both dest and src promoted structs?
             if (destDoFldAsg && srcDoFldAsg)
             {
-                // Both structs should be of the same type, or each have a single field of the same type.
+                // Both structs should be of the same type, or have the same number of fields of the same type.
                 // If not we will use a copy block.
-                if (lvaTable[destLclNum].lvVerTypeInfo.GetClassHandle() !=
-                    lvaTable[srcLclNum].lvVerTypeInfo.GetClassHandle())
+                bool misMatchedTypes = false;
+                if (destLclVar->lvVerTypeInfo.GetClassHandle() != srcLclVar->lvVerTypeInfo.GetClassHandle())
```

Comment:
Could you give me an example when do we generate `ASG` between structs with different struct handles?

Reply:
There are at least two ways this can happen:
- Generics "mismatch". The case that first called my attention to this was a mismatch between `System.Span<Canon>` and `System.Span<String>` in an assignment due to an inline.
- You can have a struct that's nested within another struct in which it's the only field.

There were about 1500 instances of this on linux/x64 I believe.

Reply:
Thank you for the explanation.